File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/PowerupFactory.java
package com.poash.rd;

public class PowerupFactory {

	public static Powerup CreatePowerup(int key) {
		switch(key) {
		case -1:
			return new NoPowerup() ;
		case 0:
			return new HealthKit() ;
		case 1:
			return new Armor() ;
		}
		return null ;
	}
}

File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/PowerupFactory.cpp
#include "PowerupFactory.h"


#include "Armor.h"
#include "HealthKit.h"
#include "NoPowerup.h"

RecklessDriver::Powerup* RecklessDriver::PowerupFactory::CreatePowerup(int key) {
	switch(key) {
	case -1:
		return new NoPowerup{} ;
	case 0:
		return new HealthKit{} ;
	case 1:
		return new Armor{} ;
	}
	return nullptr ;
}
